bundles/org.openhab.core.addon/src/main/java/org/openhab/core/addon/AddonInfoRegistry.java

```diff
@@ -52,79 +51,75 @@ protected void removeAddonInfoProvider(AddonInfoProvider addonInfoProvider) {
     }
 
     /**
-     * Returns the add-on information for the specified add-on ID, or {@code null} if no add-on information could be
+     * Returns the add-on information for the specified add-on UID, or {@code null} if no add-on information could be
      * found.
      *
-     * @param id the ID to be looked
+     * @param uid the UID to be looked
      * @return a add-on information object (could be null)
      */
-    public @Nullable AddonInfo getAddonInfo(String id) {
-        return getAddonInfo(id, null);
+    public @Nullable AddonInfo getAddonInfo(String uid) {
+        return getAddonInfo(uid, null);
     }
 
     /**
-     * Returns the add-on information for the specified add-on ID and locale (language),
+     * Returns the add-on information for the specified add-on UID and locale (language),
      * or {@code null} if no add-on information could be found.
      * <p>
-     * If more than one provider provides information for the specified add-on ID and locale,
-     * it returns merged information from all such providers.
+     * If more than one provider provides information for the specified add-on UID and locale,
+     * it returns a new {@link AddonInfo} containing merged information from all such providers.
      *
-     * @param targetId the ID to be looked for
+     * @param uid the UID to be looked for
      * @param locale the locale to be used for the add-on information (could be null)
      * @return a localized add-on information object (could be null)
      */
-    public @Nullable AddonInfo getAddonInfo(String targetId, @Nullable Locale locale) {
-        // note: using funky code to prevent a maven compiler error
-        List<AddonInfo> addonInfos = addonInfoProviders.stream()
-                .map(p -> Optional.ofNullable(p.getAddonInfo(targetId, locale))).filter(o -> o.isPresent())
-                .map(o -> o.get()).toList();
+    public @Nullable AddonInfo getAddonInfo(String uid, @Nullable Locale locale) {
+        return addonInfoProviders.stream().map(p -> p.getAddonInfo(uid, locale))
```

Comment:
Be careful. I had a runtime error without `filter(Objects::nonNull)`.

Reply:
> I had a runtime error without filter(Objects::nonNull)

Hmm. It was giving errors when trying to force the generic result type `AddonInfo` .. but changing the generic result type to `@Nullable AddonInfo` makes the build errors go away. And since the enclosing method allows to return null there _should_ be no problem.

However I still have not figured out how to write a JUnit test for validating this function. :(

Reply:
But I will add `filter(Objects::nonNull)` anyway since it really cannot do any harm. :)

Reply:
PS I am writing a Junit test

Reply:
Done.